_content/language-and-grammar.md

```diff
@@ -1,190 +1,388 @@
+- Reserve the future tense for events that will happen after a significant delay.
+_________
 
-| Recommended                       | Not recommended                         |
-| :-------------------------------- | :-------------------------------------- |
-| **This method gets the user ID.** | This method is used to get the user ID. |
+Use the present tense when writing your technical documentation, as it creates concise sentences and provides a tone of immediacy, as if things are occuring sequentially at the moment they are read (and always will). It also conveys a sense of truth or fact. 
 
-### Direct address or imperative mood
+| Do this...          | ...instead of this      |
+| :------------------- | :------------------- |
+| Click on the Open button. A pop-up window opens. | Once you have clicked on the Open button, a pop-up window will open. |
 
-Use the imperative mood for task steps.
+#### Exceptions to using the present tense
+The past tense is used for events that have happened definitively in the past and are unlikely to repeat. For Agora technical documenation, this is generally reserved for two situations: release dates and fixed issues.
 
-The imperative mood keeps the content concise. The direct address is more personal.
+Always frame a product's release date in the past tense, as the present tense is only correct on the day of release. Fixed issues described in release notes should also be written in the past tense to emphasize that the issues have been addressed.
 
-| Recommended                                          | Not recommended                           |
-| :--------------------------------------------------- | :---------------------------------------- |
-| **Download the Agora SDK. (Imperative Mood)**        | The Agora SDK can be downloaded.          |
-| **You can download the Agora SDK. (Direct Address)** | The developer can download the Agora SDK. |
+| Do this...          | ...instead of this      |
+| :------------------- | :------------------- |
+| v3.0.0 was released on Mar 4, 2020. | v3.0.0 is released on Mar 4, 2020. |
+| Disconnecting the microphone caused the audio to freeze. | Disconnecting the microphone causes the audio to freeze. |
+
+Avoid phrasing using *will* and *can*. //TODO William Why are you singling these out? "Will" is attached to the future tense, and "can" is conditional--perhaps there are occasions when it applies?
+
+Use the future tense only if there is a significant time delay that matters in context. For example, "Agora will grant you access to the service once your application has been approved." //TODO William I'm not sure about this example. Is there a better one?
+
+The present prefect tense is used to emphasize a past event that has consequences that continue into the present. For example, "Get the information from the log file if error logging has been activated." The corresponding Chinese is 已经. //TODO William The Chinese translates to "already"--is this useful here?
```

Comment:
As with all verb tenses, the present perfect is appropriate when it is appropriate.  :-)  I think a general rule to use the present tense whenever possible, but to be allowed to use the past, present perfect, or even future tenses when necessary is the best guideline. A blanket rule to always use the present tense and nothing else is too constricting, because there will always be situations where that sounds awkward. I can try to write this section to make that more clear. Is that okay?

Reply:
Yes, please.